sencha-workspace/packages/slate-cbl/src/view/teacher/DashboardController.js

```diff
@@ -222,11 +242,145 @@ Ext.define('Slate.cbl.view.teacher.DashboardController', {
             });
         });
     },
+    
+    onDemonstrationDelete: function(demonstration, operation) {
+        var me = this,
+            dashboardView = me.view,
+            mainGridEl = dashboardView.el.down('.cbl-grid-main'),
+            studentId = demonstration.get('StudentID'),
+            competenciesStore = Ext.getStore('cbl-competencies-loaded'),
+            skillStudentsTpl = dashboardView.getTpl('skillStudentsTpl'),
+
+            deletedSkills = demonstration.get('Skills'),
+            deletedSkillsLength = deletedSkills.length, deletedSkillIndex = 0, demonstratedSkill,
+            loadedCompetency, loadedSkill, demonstrationsByStudent, skillDemonstrationsCell, deletedSkillIds, studentDemonstrations, studentDemonstrationIndex = 0,
+
+            competencyCompletions = operation.getProxy().getReader().rawData.data[0].competencyCompletions,
+            competencyCompletionsLength = competencyCompletions.length, competencyCompletionIndex = 0, competencyCompletion,
+            competencyProgressCell, competencyPercent,
+            _compentencyDemonstrationResetFn,
+            _updateCompetencySkillsFn, 
+
+            updatesCompetencies = [],
+            updatesLength, updateIndex, update, skillsRows, skillHeadersRow, skillStudentsRow, competencySkills, skill,
+            
+            updatedDemonstrationLength, updatedDemonstrationIndex;
+
+
+        // compile operations and DOM references for needed updates -- don't write to dom!
+        for (; deletedSkillIndex < deletedSkillsLength; deletedSkillIndex++) {
+            loadedSkill = deletedSkills[deletedSkillIndex];
+            // update rendered demonstrations
+            skillDemonstrationsCell = mainGridEl.down('.cbl-grid-demo[data-demonstration-skill="'+loadedSkill.ID+'"]');
+
+            if (skillDemonstrationsCell) {
+                skillDemonstrationsCell.replaceCls('cbl-grid-demo-counted', 'cbl-grid-demo-empty');
+                skillDemonstrationsCell.replaceCls('cbl-grid-demo-uncounted', 'cbl-grid-demo-empty');
+                
+                skillDemonstrationsCell.update('');
+            }
+        }
+        
+        for (; competencyCompletionIndex < competencyCompletionsLength; competencyCompletionIndex++) {
+            competencyCompletion = competencyCompletions[competencyCompletionIndex];
+            competencyProgressCell = mainGridEl.down('.cbl-grid-progress-row[data-competency="'+competencyCompletion.CompetencyID+'"] .cbl-grid-progress-cell[data-student="'+studentId+'"]', true);
+            loadedCompetency = competenciesStore.getById(competencyCompletion.CompetencyID);
+            
+            if (!competencyProgressCell || !loadedCompetency) {
+                continue;
+            }
+
+            competencyPercent = Math.round(100 * (competencyCompletion.demonstrationsCount || 0) / loadedCompetency.get('totalDemonstrationsRequired'));
```

Comment:
I don't think so, DemonstrationsRequired for any given skill must be >= 1 and there shouldn't be any competencies with no skills